### PR TITLE
Add dev.bsnes.bsnes

### PR DESCRIPTION
**Pull request by ghisvail**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

Requires `--device=all` for controller access.

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by ghisvail** [comment]
I still have to find a way to provide saner defaults on first launch, i.e. use OpenGL rendering and Pulseaudio mixing.

File: dev.bsnes.bsnes.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Copyright 2021 Flathub maintainers -->
+<component type="desktop-application">
+<id>dev.bsnes.bsnes</id>
+  <id>dev.ares.ares</id>
```
Comment (Screwtapello) [inline dev.bsnes.bsnes.metainfo.xml:5]: Is that supposed to be like that?

Comment (ghisvail) [reply dev.bsnes.bsnes.metainfo.xml:5]: My bad, classic copy pasta error

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

File: dev.bsnes.bsnes.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+app-id: dev.bsnes.bsnes
+runtime: org.gnome.Platform
```
Comment (hfiguiere) [inline dev.bsnes.bsnes.yaml:2]: Do you really need the GNOME runtime? Isn't the freedesktop SDK enough?

Comment (ghisvail) [reply dev.bsnes.bsnes.yaml]: Indeed, the freedesktop runtime is enough :+1:

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

**Comment by ghisvail** [comment]
bot, build dev.bsnes.bsnes

**Comment by ghisvail** [comment]
This is now ready for review I think.

**Pull request closed without merging**